### DISCUSSION TITLE
Int Parser on just "-"

### DISCUSSION THREAD

**Discussion by mortigenus** [pointfreeco/swift-parsing#36] [Q&A] [answered]
Is it expected for an Int parser to return 0 when the input is just "-"? I would expect parser to fail unless there's a digit after the minus sign, but I wanted to know what others think before posting it as an issue.
I tripped on this while trying to parse a math expression where I have a literal "-" parser and an Int parser combined in `OneOf`.

**Comment by mbrandonw** [comment, marked as answer]
Hey @mortigenus, thanks for the report! I think this is a bug because `Int`'s initializer does not parse `"-"`: `Int("-") == nil`. 😕

We'll fix it soon or if you have time to give it a shot feel free!

**Reply by mortigenus** [reply]
Thanks for your reply, I've cooked up a suggestion https://github.com/pointfreeco/swift-parsing/pull/37
